aiida_vasp/parsers/file_parsers/vasprun.py

```diff
@@ -510,12 +511,18 @@ def _energies(self, nosc):
                 self._vasp_parser.exit_code = self._vasp_parser.exit_codes.ERROR_NOT_ABLE_TO_PARSE_QUANTITY.format(
                     quantity=str(sys._getframe().f_code.co_name))
                 return None
-            # should be a list, but convert to ndarray, here
-            # staggered arrays are not a problem
-            # two elements for a static run, both are similar,
-            # only take the last
-            if len(enrgies) == 2:
-                enrgies = enrgies[-1:]
+
```

Comment:
Actually, I can't understand the expected behaviour of this part and what the comments mean.

Reply:
Sorry, I will try to be more clear. In fact, this review comment was supposed to go in right after you sent the PM, but I somehow did not press review.

> On `store_sc_energies`, I have no idea how to add this option.

The code snippet was related to this. To show how we can utilize setting e.g. `parser_settings: {store_sc_energies: True}` in the `_energies` routine we place a code similar to the one above and put whatever needs to be put under each case. 

> And update `DEFAULT_OPTIONS`. The `settings` here is the content of `parser_settings` that we supply to the calculation `settings`.

We need to add the default of `store_sc_energies` to `DEFAULT_OPTIONS`.

> I think it is possible best to keep the structure and remove the list collapse so that it is the same regardless of approach. This would also clearly indicate that only one ionic step was performed. What do you think?

This was related to if we should collapse the index when we do not have sc steps. I would say no. E.g. we have always `energies: np.array(ionic_steps, sc_steps)` and not `energies: np.array(ionic_steps)` for the no sc case.

Reply:
Thanks for your explanation. I understand your sentence for `store_sc_energies` 

On the last lines, we have to be very sure about the data structure.  `ArrayData` can store multiple arrays. What I was considering is that we keep current `energies`, and just adding two new one dimensional arrays. One is the flattened (concatenated) eneriges for all ionic steps and all SCF steps. The other is the numbers of SCF steps for all ionic steps. Do I explain what I think well?

Reply:
Agreed. So in the case of not having sc steps, we do not set these extra arrays and if we enable it, we still set the present `energies` so that we have somewhat of a double storage, or do you then want to not store this?

Reply:
I'm sorry, my explanation was bad. Currently with ionic relaxation, the `node.outputs.energies` is like this
```
In [7]: n.outputs.energies.get_array('energy_no_entropy')
Out[7]:
array([-31.80518222, -31.80443243, -31.81059057, -31.81059058,
       -31.81059116, -31.81059225, -31.8105924 , -31.81059241,
       -31.81059241, -31.81059241])
```
This is the last energies of the ionic steps as far as I understand. We can keep this as the digest information. In addition, we can have the detailed energy information in the same node, which is something like
```
n.outputs.energies.get_array('energy_no_entropy_sc')
```
The latter contains the former data, but often their use cases are considered different. And the concatenated data of the latter requires one more step to extract the former data from the latter. So we can keep former even with when having the latter. Once we decide to have the latter, the former data size is negligible. So it's OK to have both in terms of data size.

Reply:
Okey, then let us do this. We keep the default to being the last energy for each step and then open for `energy_type_sc` in addition to `energy_type` energy entries in the array. I agree that if we enable `energy_type_sc` the size of `energy_type_sc` is usually negligible. Then by default we do not disable this. I suggest we enable it with `parser_settings: {'store_energies_sc': bool}` with some default set as above. We can of course change `store_energies_sc` flag to something else. Do you want to update the PR Togo or should I do it?

Reply:
Sure, I will update it, which will include the update to store multiple arrays in an `ArrayData` node.

Reply:
Thanks a lot Togo. Greatly appreciated.

Reply:
@atztogo Do you have time to update this or should I do it? Thanks.